### PR TITLE
feat: Add Rpc skeleton

### PR DESCRIPTION
**Pull request by Elyniss**
Merge after #786

### CODE REVIEW

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/responses/ResponseUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.walletconnect.sync.engine.use_case.responses
```
Comment (TalhaAli00) [inline sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/responses/ResponseUseCase.kt:1]: same as above

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/domain/SyncEngine.kt [hunk 2]
```diff
@@ -1,29 +1,102 @@
...
+                    is SyncParams.SetParams -> onSetResponseUseCase(params, response)
+                    is SyncParams.DeleteParams -> onDeleteResponseUseCase(params, response)
+                }
+            }
+        }
+
+    private fun collectInternalErrors(): Job =
+        merge(jsonRpcInteractor.internalErrors, pairingHandler.findWrongMethodsFlow)
+            .onEach { exception -> _events.emit(exception) }
+            .launchIn(scope)
```
Comment (jakubuid) [inline sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/domain/SyncEngine.kt:101]: This class is so beautiful with UseCases. Can't wait to refactor Sign and Auth

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/RequestUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.walletconnect.sync.engine.use_case.requests
```
[the remark this answers is not in the document]
Comment (kacperoak) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/RequestUseCase.kt:1]: I agree with the team, I do not see the benefits of these interfaces

[the remark this answers is not in the document]
Comment (Elyniss) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/RequestUseCase.kt:1]: Cool, will remove them

**Review by jakubuid — APPROVED** [review]

**Pull request merged**